internal/hcsoci/create.go

```diff
@@ -236,8 +235,19 @@ func configureSandboxNetwork(ctx context.Context, coi *createOptionsInternal, r
 		// container but not a workload container in a sandbox that inherits
 		// the namespace.
 		if ct == oci.KubernetesContainerTypeNone || ct == oci.KubernetesContainerTypeSandbox {
-			if err := SetupNetworkNamespace(ctx, coi.HostingSystem, coi.actualNetworkNamespace); err != nil {
-				return err
+			if err := coi.HostingSystem.ConfigureNetworking(ctx, coi.actualNetworkNamespace); err != nil {
+				// No network setup type was specified for this UVM. Create and assign one here unless
+				// we received a different error.
+				if err == uvm.ErrNoNetworkSetup {
+					if err := coi.HostingSystem.CreateAndAssignNetworkSetup(ctx, "", ""); err != nil {
```

Comment:
do we know for sure at this point that ncproxy is not enabled for the VM?

Reply:
The only time in our use cases we'd end up here is for a standalone task (launched through ctr) and we don't have any code to setup ncproxy for that so I think this is a fine assumption for now.